### PR TITLE
[X86][EVEX512] Do not allow 512-bit memcpy without EVEX512

### PR DESCRIPTION
**Pull request by phoebewang**
Solves crash mentioned in #65920.

### CODE REVIEW

File: llvm/test/CodeGen/X86/evex512-mem.ll [hunk 1]
```diff
@@ -0,0 +1,29 @@
+; NOTE: Assertions have been autogenerated by utils/update_llc_test_checks.py
+; RUN: llc -mtriple=x86_64-apple-darwin -mattr=avx512f,avx512bw,avx512vl < %s | FileCheck %s --check-prefix=AVX512
```
Comment (KanRobert) [inline llvm/test/CodeGen/X86/evex512-mem.ll:2]: -mtriple=x86_64-apple-darwin -> -mtriple=x86_64-unknown ?

File: llvm/lib/Target/X86/X86ISelLoweringCall.cpp [hunk 2]
```diff
@@ -281,7 +281,7 @@ EVT X86TargetLowering::getOptimalMemOpType(
     if (Op.size() >= 16 &&
         (!Subtarget.isUnalignedMem16Slow() || Op.isAligned(Align(16)))) {
       // FIXME: Check if unaligned 64-byte accesses are slow.
-      if (Op.size() >= 64 && Subtarget.hasAVX512() &&
+      if (Op.size() >= 64 && Subtarget.hasAVX512() && Subtarget.hasEVEX512() &&
           (Subtarget.getPreferVectorWidth() >= 512)) {
```
Comment (KanRobert) [inline llvm/lib/Target/X86/X86ISelLoweringCall.cpp:285]: Should we update `X86TTIImpl::getRegisterBitWidth` too?

File: llvm/test/CodeGen/X86/evex512-mem.ll (hunk 1 above)
Comment (phoebewang) [reply llvm/test/CodeGen/X86/evex512-mem.ll]: Done.

File: llvm/lib/Target/X86/X86ISelLoweringCall.cpp (hunk 2 above)
Comment (phoebewang) [reply llvm/lib/Target/X86/X86ISelLoweringCall.cpp:285]: Good idea, done.

**Review by KanRobert — APPROVED** [review]
LGTM

**Pull request merged**